tools/swift-stdlib-tool/swift-stdlib-tool.cpp

```diff
@@ -670,583 +643,576 @@ -(NSString *)sst_command {
+    dispatch_release(gotStdErr);
+
+    int status = 0;
+    waitpid(childPid, &status, 0);
+    
+    // Task is finished and we have its stdout and stderr output.
+
+    // Print its stdout and stderr if it failed or we are verbose.
+    // (Print nothing by default because codesign is noisy.)
+
+    if (status  ||  Verbose > 1) {
+        fwrite(stdErrData.data(), stdErrData.size(), 1, stdout);
+        fwrite(stdOutData.data(), stdOutData.size(), 1, stdout);
     }
+    if (block) { block(stdOutData, stdErrData, status); }
+    return status;
 }
 
 
-void copyAndStripBitcode(NSString *src, NSString *dst)
+void copyAndStripBitcode(std::filesystem::path src, std::filesystem::path dst)
 {
     // -r removes bitcode
-    int err = xcrunToolCommand(@[@"bitcode_strip", src, @"-r", @"-o", dst]);
+    std::vector<std::string> commandAndArgs = {"bitcode_strip", src, "-r", "-o", dst};
+    int err = xcrunToolCommand(commandAndArgs);
 
     // Fail if bitcode_strip failed.
     if (err) {
         fail("Couldn't copy and strip bitcode %s to %s: bitcode_strip failed "
-             "with exit code %d", 
-             src.fileSystemRepresentation, dst.fileSystemRepresentation, err);
+             "with exit code %d", src.c_str(), dst.c_str(), err);
     }
 }
 
 
 void 
-copyFile(NSFileManager *fm, NSString *src, NSString *dst, bool stripBitcode)
+copyFile(std::filesystem::path src, std::filesystem::path dst, bool stripBitcode)
 {
     if (stripBitcode) {
         copyAndStripBitcode(src, dst);
     } else {
-        NSError *nserr;
-        if (![fm copyItemAtPath:src toPath:dst error:&nserr]) {
-            fail("Couldn't copy %s to %s: %s", 
-                 src.fileSystemRepresentation, 
-                 dst.fileSystemRepresentation, 
-                 nserr.localizedFailureReason.UTF8String);
+        if (rename(src.c_str(), dst.c_str()) != 0) {
+            fail("Couldn't copy %s to %s: %s", src.c_str(), dst.c_str(), strerror(errno));
         }
-    }
+   }
 }
 
+std::string uuidString(uuid_t uuid) {
+    char buffer[37];
+    uuid_unparse(uuid, buffer);
+    return buffer;
+}
 
-void copyLibraries(NSString *src_dir, NSString *dst_dir, 
-                   NSMutableDictionary *libs, bool stripBitcode)
+void copyLibraries(std::filesystem::path src_dir, std::filesystem::path dst_dir, 
+                   std::unordered_map<std::string,
+                   std::unordered_set<std::string>> libs,
+                   bool stripBitcode)
 {
-    NSFileManager *fm = NSFileManager.defaultManager;
-
-    [fm createDirectoryAtPath: dst_dir withIntermediateDirectories:YES 
-     attributes:nil error:nil];
+    mkpath_np(dst_dir.c_str(), S_IRWXU | S_IRWXG | S_IRWXO);
     
-    for (NSString *lib in libs) @autoreleasepool {
-        NSString *src = [src_dir stringByAppendingPathComponent:lib]; 
-        NSString *dst = [dst_dir stringByAppendingPathComponent:lib];
+    for (auto const &pair : libs) {
```

Comment:
As long as we're using C++17:
```suggestion
    for (auto const &[lib, srcUUIDs] : libs) {
```

Reply:
Thanks, I didn't know this was possible in C++